### ISSUE TITLE
Object conflicts and failed checksums questions

### ISSUE THREAD

**Issue by mrtnptrs** [OpenRCT2/OpenRCT2#4063]
**OS:** Windows 10
**Version:** 0.0.5
**Commit/Build:** a4223d9

I have a few questions regarding the output at the commandprompt window. I have asked this before, but I really got no good answer. And a week ago I asked it at the OpenRCT2.org site and also got no reply, well... actually one, which said me to ask it here (was already planning to do it). 

**Steps to reproduce:**
1. Delete everything at the OpenRCT2 documents folder.
2. Re-download OpenRCT2 and start it.
3. Point it to the RCT2 installation directory.
4. See the magic happening at the commandprompt window.

**This is the commandprompt output:**
```
Scanning 4762 objects...
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\1KEGYFPI.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\1KEGYFPJ.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\aswfd.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\ASWFDA.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\BALLN.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\BALLO.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\BIGBEN.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\BIGBEO.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\COF4MKY.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\COF4MKZ.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\DRAGON.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\DRAGOO.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\hrs.dat'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\HRT.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\JATLPTHT.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\JATLPTHU.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\PATHRDRT.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\PATHRDRU.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\patrctfan#dp.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\patrctfan#snd.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\patrctfan#dp.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\patrctfan#st.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\STONE31.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\STONE34.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\TTRFGL01.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\TTRFGL03.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\AWIN3A.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\Tunnelb.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\AWIN1A.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\Tunnels.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\AWIN2A.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\Tunnelt.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\WALLWF33.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\Wallwf34.dat'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\WALLWF33.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\WALLWF35.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\WATER.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\WATES.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\XXBSHC01.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\XXBSHC02.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\XXTRCY01.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\XXTRCY02.DAT'
Scanning complete in 1.10 seconds.
ERROR[src\ride\track_design.c:99 (track_design_open)]: Track checksum failed.
ERROR[src\ride\track_design.c:99 (track_design_open)]: Track checksum failed.
ERROR[src\ride\track_design.c:99 (track_design_open)]: Track checksum failed.
ERROR[src\ride\track_design.c:99 (track_design_open)]: Track checksum failed.
ERROR[src\ride\track_design.c:99 (track_design_open)]: Track checksum failed.
ERROR[src\ride\track_design.c:99 (track_design_open)]: Track checksum failed.
ERROR[src\ride\track_design.c:99 (track_design_open)]: Track checksum failed.
```

**Object conflict?**

Do the above object conflict errors mean that I have some data files that have a different name, but are actually the same (and a different timestamp)? Can I "fix" this by just deleting one of the two object files without losing any objects? So, for example that 1KEGYFPI.DAT and 1KEGYFPJ.DAT are in conflict and I can just choose to delete one of them, as they are the same object actually?

**Track checksum failed**

Ok, this means that I have a few track files of which the checksum does not match with the calculated checksum by OpenRCT2, right? So, that means that I have 7 track-files which are corrupted or hacked. But the output does not mention the names of the tracks with the failed checksum, making this output completely useless. Is there a way for you to add the name or path of the "broken" track-files to the errors at the output of the commandprompt, so that I can look at these files and can delete them, so they won't cause crashes in OpenRCT2 and RCT2 vanilla?

**Another question**

Why do the tracksum failed errors show at every restart at the commandprompt, but the Object conflict only at first start? So, is it true then that the checksums of the ride files (maybe also the scenario files) get checked by every startup, but the object scanning only at first startup? When yes, why is that?

**Comment by janisozaur** [comment]
@mrtnptrs if you have a unch what is happening ("I have some data files that have a different name, but are actually the same") why won't you check these files' hashes?

The track names should be displayed when you launch with `--verbose`

**Comment by duncanspumpkin** [comment]
The object conflict is caused by badly made objects. If you either delete both or the one with the higher character number it should all work fine.

**Comment by IntelOrca** [comment]
Its possible that your object conflicts were created due to a bug in the game which has since been fixed. Its very likely that they are indeed duplicates, but as @janisozaur suggested, you can hash the conflicts to confirm.

**Comment by mrtnptrs** [comment]
@janisozaur Because I'm not an expert with that, but I could Google about hashes and check them by hand OR I could ask you, as you are the experts and @IntelOrca made the code. :)

And NO, I tried it many times using the --verbose "tag", but it does not show the affected track files.

@duncanspumpkin Do you mean that these object files are BOTH probably corrupted or just duplicates as @IntelOrca said? I will check their hashes and report back to you about it.

Also, using --verbose, I saw more interesting output:

```
Scanning 4762 objects...
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\1KEGYFPI.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\1KEGYFPJ.DAT'
VERBOSE: [1KMTRF52] Warning: Image table size longer than expected.
VERBOSE: [1KMTXXRF] Warning: Image table size longer than expected.
VERBOSE: [1KSBRLL1] Warning: Image table size longer than expected.
VERBOSE: [1KSBRLL2] Warning: Image table size longer than expected.
VERBOSE: [1KSBRLL3] Warning: Image table size longer than expected.
VERBOSE: [1KXXMB20] Warning: Image table size longer than expected.
VERBOSE: [1KXXMB21] Warning: Image table size longer than expected.
VERBOSE: [AE-HMBTB] Warning: Image table size longer than expected.
VERBOSE: [ARLA01  ] Warning: Image table size longer than expected.
VERBOSE: [ARLA03  ] Warning: Image table size longer than expected.
VERBOSE: [ARLA05  ] Warning: Image table size longer than expected.
VERBOSE: [ARMI26  ] Warning: Image table size longer than expected.
VERBOSE: [ARRT3   ] Warning: Image table size longer than expected.
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\aswfd.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\ASWFDA.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\BALLN.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\BALLO.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\BIGBEN.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\BIGBEO.DAT'
VERBOSE: [BMCMTCR1] Warning: Image table size longer than expected.
VERBOSE: [BMCMTCR2] Warning: Image table size longer than expected.
VERBOSE: [BMCMTCR3] Warning: Image table size longer than expected.
VERBOSE: [BMDIVE10] Warning: Image table size longer than expected.
VERBOSE: [BMDIVE8 ] Warning: Image table size longer than expected.
VERBOSE: [BMWING  ] Warning: Image table size longer than expected.
VERBOSE: [BOULDJNG] Warning: Image table size longer than expected.
VERBOSE: [CJK-TR8 ] Warning: Image table size longer than expected.
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\COF4MKY.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\COF4MKZ.DAT'
VERBOSE: [D-SB01  ] Warning: Image table size longer than expected.
VERBOSE: [D0000099] Warning: Image table size longer than expected.
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\DRAGON.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\DRAGOO.DAT'
VERBOSE: [GWBMCRMT] Warning: Image table size longer than expected.
VERBOSE: [HOUSRCK2] Warning: Image table size longer than expected.
VERBOSE: [HOUSROCK] Warning: Image table size longer than expected.
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\hrs.dat'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\HRT.DAT'
VERBOSE: [INTWING ] Warning: Image table size longer than expected.
VERBOSE: [ITMCANVD] Warning: Image table size longer than expected.
VERBOSE: [ITMCANVS] Warning: Image table size longer than expected.
VERBOSE: [JAGIGLT ] Warning: Image table size longer than expected.
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\JATLPTHT.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\JATLPTHU.DAT'
VERBOSE: [JCBNCH1 ] Warning: Image table size shorter than expected.
VERBOSE: [KHDVBRD2] Warning: Image table size longer than expected.
VERBOSE: [KHHDRAIL] Warning: Image table size longer than expected.
VERBOSE: [KNGBRP2B] Warning: Image table size longer than expected.
VERBOSE: [KNGBRPA2] Warning: Image table size longer than expected.
VERBOSE: [KNGCODPA] Warning: Image table size longer than expected.
VERBOSE: [KNGCOLPA] Warning: Image table size longer than expected.
VERBOSE: [KNGDBRP2] Warning: Image table size longer than expected.
VERBOSE: [KNGDHPDP] Warning: Image table size longer than expected.
VERBOSE: [KNGHPDPA] Warning: Image table size longer than expected.
VERBOSE: [KNGTCGRF] Warning: Image table size longer than expected.
VERBOSE: [LAVA20  ] Warning: Image table size longer than expected.
VERBOSE: [LAVA21  ] Warning: Image table size longer than expected.
VERBOSE: [LITBRU  ] Warning: Image table size shorter than expected.
VERBOSE: [LMPDFNC1] Warning: Image table size longer than expected.
VERBOSE: [LVSUPP01] Warning: Image table size longer than expected.
VERBOSE: [LVSUPP02] Warning: Image table size longer than expected.
VERBOSE: [LVSUPP05] Warning: Image table size longer than expected.
VERBOSE: [LVSUPP07] Warning: Image table size longer than expected.
VERBOSE: [LVSUPP08] Warning: Image table size longer than expected.
VERBOSE: [LVSUPP09] Warning: Image table size longer than expected.
VERBOSE: [LVSUPP0A] Warning: Image table size longer than expected.
VERBOSE: [LVSUPP0B] Warning: Image table size longer than expected.
VERBOSE: [LVSUPP0C] Warning: Image table size longer than expected.
VERBOSE: [LVSUPP0D] Warning: Image table size longer than expected.
VERBOSE: [LVSUPP0E] Warning: Image table size longer than expected.
VERBOSE: [LVSUPP0F] Warning: Image table size longer than expected.
VERBOSE: [MACKMEGA] Warning: Image table size longer than expected.
VERBOSE: [MGBMT1X1] Warning: Image table size longer than expected.
VERBOSE: [MGCBG3X3] Warning: Image table size longer than expected.
VERBOSE: [MGCBG5X5] Warning: Image table size longer than expected.
VERBOSE: [MGCBG7X7] Warning: Image table size longer than expected.
VERBOSE: [MGMTPHCZ] Warning: Image table size longer than expected.
VERBOSE: [MGMTPHDT] Warning: Image table size longer than expected.
VERBOSE: [MGMTPHTC] Warning: Image table size longer than expected.
VERBOSE: [MGTOT1  ] Warning: Image table size longer than expected.
VERBOSE: [MGTOT2  ] Warning: Image table size longer than expected.
VERBOSE: [MSSPIN  ] Warning: Image table size longer than expected.
VERBOSE: [MTHANG1 ] Warning: Image table size longer than expected.
VERBOSE: [MTMMENT1] Warning: Image table size longer than expected.
VERBOSE: [MTMMENT2] Warning: Image table size longer than expected.
VERBOSE: [MTMMENT3] Warning: Image table size longer than expected.
VERBOSE: [MTTBN1  ] Warning: Image table size longer than expected.
VERBOSE: [MTTCT1  ] Warning: Image table size longer than expected.
VERBOSE: [MTTCT2  ] Warning: Image table size longer than expected.
VERBOSE: [MTTDN4  ] Warning: Image table size longer than expected.
VERBOSE: [MTTDN5  ] Warning: Image table size longer than expected.
VERBOSE: [MTTEPEE ] Warning: Image table size longer than expected.
VERBOSE: [MTTGC1  ] Warning: Image table size longer than expected.
VERBOSE: [MTTGC2  ] Warning: Image table size longer than expected.
VERBOSE: [MTTGS   ] Warning: Image table size longer than expected.
VERBOSE: [MTTL3   ] Warning: Image table size longer than expected.
VERBOSE: [MTTM2   ] Warning: Image table size longer than expected.
VERBOSE: [MTTMJ   ] Warning: Image table size longer than expected.
VERBOSE: [MTTMM1  ] Warning: Image table size longer than expected.
VERBOSE: [MTTOH1  ] Warning: Image table size longer than expected.
VERBOSE: [MTTOH2  ] Warning: Image table size longer than expected.
VERBOSE: [MTTOTEM ] Warning: Image table size longer than expected.
VERBOSE: [MTTT1   ] Warning: Image table size longer than expected.
VERBOSE: [MTUMBRLL] Warning: Image table size longer than expected.
VERBOSE: [MTWAG1  ] Warning: Image table size longer than expected.
VERBOSE: [MTWAG2  ] Warning: Image table size longer than expected.
VERBOSE: [MVBURR  ] Warning: Image table size longer than expected.
VERBOSE: [MVCOLJAG] Warning: Image table size longer than expected.
VERBOSE: [MVGIANRC] Warning: Image table size longer than expected.
VERBOSE: [MVGIANTR] Warning: Image table size longer than expected.
VERBOSE: [MVICEBE3] Warning: Image table size longer than expected.
VERBOSE: [MVICEBE4] Warning: Image table size longer than expected.
VERBOSE: [MVJAGGED] Warning: Image table size longer than expected.
VERBOSE: [MVRCAVE ] Warning: Image table size longer than expected.
VERBOSE: [MVRFORM1] Warning: Image table size longer than expected.
VERBOSE: [MVRFORM2] Warning: Image table size longer than expected.
VERBOSE: [MVSNORCA] Warning: Image table size longer than expected.
VERBOSE: [NESU15  ] Warning: Image table size longer than expected.
VERBOSE: [NESUD14 ] Warning: Image table size longer than expected.
VERBOSE: [NESUD15 ] Warning: Image table size longer than expected.
VERBOSE: [NESUD16 ] Warning: Image table size longer than expected.
VERBOSE: [NESUFF11] Warning: Image table size longer than expected.
VERBOSE: [NESUFH11] Warning: Image table size longer than expected.
VERBOSE: [NESUVC12] Warning: Image table size longer than expected.
VERBOSE: [NESUVC13] Warning: Image table size longer than expected.
VERBOSE: [NESUVF11] Warning: Image table size longer than expected.
VERBOSE: [NESUVF12] Warning: Image table size longer than expected.
VERBOSE: [NESUVH11] Warning: Image table size longer than expected.
VERBOSE: [NESUVH12] Warning: Image table size longer than expected.
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\PATHRDRT.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\PATHRDRU.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\patrctfan#dp.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\patrctfan#snd.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\patrctfan#dp.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\patrctfan#st.DAT'
VERBOSE: [RMCT1   ] Warning: Image table size longer than expected.
VERBOSE: [RSWR02MT] Warning: Image table size longer than expected.
VERBOSE: [RSWR03MT] Warning: Image table size longer than expected.
VERBOSE: [SF-DRF  ] Warning: Image table size longer than expected.
VERBOSE: [SF-DRF2 ] Warning: Image table size longer than expected.
VERBOSE: [SF2X2TRE] Warning: Image table size longer than expected.
VERBOSE: [SFBENCH1] Warning: Image table size longer than expected.
VERBOSE: [SFBENCH6] Warning: Image table size longer than expected.
VERBOSE: [SFBIN6  ] Warning: Image table size longer than expected.
VERBOSE: [SFLAMP3 ] Warning: Image table size longer than expected.
VERBOSE: [SFLAMP5 ] Warning: Image table size longer than expected.
VERBOSE: [SFPBMT6 ] Warning: Image table size longer than expected.
VERBOSE: [SHOP1MKY] Warning: Image table size shorter than expected.
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\STONE31.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\STONE34.DAT'
VERBOSE: [TRACK10 ] Warning: Image table size longer than expected.
VERBOSE: [TRACK11 ] Warning: Image table size longer than expected.
VERBOSE: [TRACK12 ] Warning: Image table size longer than expected.
VERBOSE: [TRACK13 ] Warning: Image table size longer than expected.
VERBOSE: [TRACK14 ] Warning: Image table size longer than expected.
VERBOSE: [TRACK15 ] Warning: Image table size longer than expected.
VERBOSE: [TRACK16 ] Warning: Image table size longer than expected.
VERBOSE: [TRACK17 ] Warning: Image table size longer than expected.
VERBOSE: [TRACK18 ] Warning: Image table size longer than expected.
VERBOSE: [TRACK19 ] Warning: Image table size longer than expected.
VERBOSE: [TRACK20 ] Warning: Image table size longer than expected.
VERBOSE: [TRACK21 ] Warning: Image table size longer than expected.
VERBOSE: [TRACK31 ] Warning: Image table size longer than expected.
VERBOSE: [TRACK32 ] Warning: Image table size longer than expected.
VERBOSE: [TRACK6  ] Warning: Image table size longer than expected.
VERBOSE: [TRACK7  ] Warning: Image table size longer than expected.
VERBOSE: [TRACK8  ] Warning: Image table size longer than expected.
VERBOSE: [TRACK9  ] Warning: Image table size longer than expected.
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\TTRFGL01.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\TTRFGL03.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\AWIN3A.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\Tunnelb.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\AWIN1A.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\Tunnels.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\AWIN2A.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\Tunnelt.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\WALLWF33.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\Wallwf34.dat'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\WALLWF33.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\WALLWF35.DAT'
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\WATER.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\WATES.DAT'
VERBOSE: [XXARCS13] Warning: Image table size longer than expected.
VERBOSE: [XXARCS15] Warning: Image table size longer than expected.
VERBOSE: [XXARCS16] Warning: Image table size longer than expected.
VERBOSE: [XXBMTR01] Warning: Image table size longer than expected.
VERBOSE: [XXBMTR02] Warning: Image table size longer than expected.
VERBOSE: [XXBMTR03] Warning: Image table size longer than expected.
VERBOSE: [XXBMTR05] Warning: Image table size longer than expected.
VERBOSE: [XXBMTR08] Warning: Image table size longer than expected.
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\XXBSHC01.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\XXBSHC02.DAT'
VERBOSE: [XXRFMT52] Warning: Image table size longer than expected.
Object conflict: 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\XXTRCY01.DAT'
               : 'E:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\ObjData\XXTRCY02.DAT'
Scanning complete in 1.10 seconds.
```

So, what does it mean that the table size of an object file is longer that expected? Just curious and maybe I can add something about it at a wikipage to help others who see these "errors" :)

**Comment by IntelOrca** [comment]
> So, what does it mean that the table size of an object file is longer that expected?

It means all of those objects were not properly written by the program used to create them. Unfortunately these object creation programs had a few bugs regarding this.

**Comment by mrtnptrs** [comment]
@duncanspumpkin Ok, I can now find the corrupt/hacked track-files and delete them if I want.

@IntelOrca So, these objects are just not properly written, but that does not (necessarily) mean they do not work/are corrupted and/or let OpenRCT2 or vanilla RCT2 crash?

Wouldn't it be a good idea to make a wiki page with all the kind of errors you can see at the commandprompt window with the explanation of it, how severe the problem could be and what actions to take best? (also mentioning the --verbose errors for people who found that out) Then it would be much more clear for people what the errors mean and then they don't get "scared" of them that quickly.

**Comment by X123M3-256** [comment]
@duncanspumpkin Do you know more specifically what causes this? I'm getting object conflicts on objects I've made, and if it's a problem with the objects or my tool I'd like to correct it.

**Comment by IntelOrca** [comment]
@X123M3-256 you get a conflict if there are two object files found with the same name (the one stored within the file).

**Comment by mrtnptrs** [comment]
@duncanspumpkin @janisozaur @IntelOrca Not to annoy you, but.... you have only answered around 75% of the questions here :) I know you have something else to do and this has no priority, but.... you get what I mean... Want me to sum up what the questions are which still "need" to be answered?

**Comment by duncanspumpkin** [comment]
You will get a conflict if the object names of two files match. Note this is not the same as the name of the file. For example both BALLN and BALLO contain as the first part of their file a header that among other things has an object name. This object name will be "BALLN   ". When writing this object to the computer it first checks to see if BALLN.dat already exists. If it exists AND its the same file it doesn't create anything. If it exists AND its a different file it will increment the last character to create BALLO.dat. The problem is BALLN and BALLO have the same object name but there is some difference in their files. This can occur when people make multiple versions of the same file or they use a buggy ride maker that writes data to a section that is later blanked out. The object exporter is designed to handle buggy objects that have modified the wrong section by creating extra data that makes it think that the files are identical. This will then prevent the creation of BALLO.dat. 

So in summary the best idea would be to delete all conflict files and reload them by loading a park that includes them.

**Comment by IntelOrca** [comment]
Just so you know, the new code will no longer export an object unless that name doesn't exist. So it shouldn't create new conflicts anymore.

**Comment by X123M3-256** [comment]
@duncanspumpkin I thought I'd made sure I didn't write any fields that later got zeroed out (doesn't this make the object unexportable as well?). Is it possible that this is caused by old versions of the objects still in circulation? Some of my rides had to be updated a while back, and they were replaced with new versions that had identical names, identical checksums, but different contents. If I were to open a park made with the older object, would it cause a duplicate to be created?

**Comment by duncanspumpkin** [comment]
@x123 I don't really know how exactly the new version works.

**Comment by mrtnptrs** [comment]
And for the object files which have the Image table size longer than expected warning when using --Verbose? What is the best way to deal with these files? Delete them? @IntelOrca said that this was caused by object creator programs which were not properly written, but how bad is it that the image table size is too long? So, what is the best action to take on this? Also deleting thse objects? @duncanspumpkin

**Comment by IntelOrca** [comment]
@X123M3-256 no it shouldn't export the object if an object with the same name (as specified in the file) already exists.

**Comment by duncanspumpkin** [comment]
Just ignore them there is nothing that wrong with them

**Comment by IntelOrca** [comment]
@duncanspumpkin well there is... :smile:

**Comment by mrtnptrs** [comment]
@IntelOrca @duncanspumpkin Isn't it then better to disable that warning in --verbose mode as there is nothing you can do about it? Or is it useful for debug purposes?

**Comment by mrtnptrs** [comment]
When above questions are answered, you can close this issue :)

**Comment by X123M3-256** [comment]
Looking at the code, it seems that the "Image table longer than expected" message is triggered when the actual size of the bitmap data does not match the size given in the file. This doesn't seem to be a serious issue, because as far as I can tell the value in the file isn't actually used much.

I do, however, think the warning should remain - verbose mode is supposed to print as much information as possible, and it's useful to know which objects are broken so you can fix them, even if this problem doesn't seem to have much effect.

**Comment by IntelOrca** [comment]
I originally didn't have "Image table longer than expected", but because objects don't store the correct image table length, I just have to assume that the remainder of the file is image table, and if there is excess bytes - something isn't quite right.

**Comment by mrtnptrs** [comment]
Then this issue can be closed now or maybe kept with the 'information'-tag.
